firestore/paginate_data.rb

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/firestore"
+
+def start_at_field_query_cursor project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  cities_ref = firestore.col "cities"
+  # [START fs_start_at_field_query_cursor]
+  query = cities_ref.order("population").start_at(1000000)
+  # [END fs_start_at_field_query_cursor]
+  query.get do |city|
+    puts "Document #{city.document_id} returned by start at population 1000000 field query cursor."
+  end
+end
+
+def end_at_field_query_cursor project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  cities_ref = firestore.col "cities"
+  # [START fs_end_at_field_query_cursor]
+  query = cities_ref.order("population").end_at(1000000)
+  # [END fs_end_at_field_query_cursor]
+  query.get do |city|
+    puts "Document #{city.document_id} returned by end at population 1000000 field query cursor."
+  end
+end
+
+def paginated_query_cursor project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  # [START fs_paginated_query_cursor]
+  cities_ref = firestore.col "cities"
+  first_query = cities_ref.order("population").limit(3)
+
+  # Get the last document from the results.
+  last_population = 0
+  first_query.get do |city|
+    last_population = city.data[:population]
```

Comment:
this is recommend by GitHub style guide (for reading hashes). But for writing hash values it is not recommended. (see above)